### PR TITLE
Add verbosity flag for MPB

### LINKED ISSUE #116: quieter output in Jupyter notebooks
By default, MPB prints a lot of information — this stems from its original mode of operation, where you typically piped all of the output to a file and `grep`-ed for the data you want.   When you run it in a Jupyter notebook however, and are obtaining the data programmatically rather than by grepping, all of this output is annoying.

It would be good to implement a "quiet" mode (i.e. `verbosity` scale similar to Meep), and perhaps turn it on by default during interactive use, to suppress most of this output.

Basically, this means:

1. Adding a `verbosity` global variable to MPB (I think we already have a `verbose` global Boolean, but it would be better to have an integer similar to Meep so that 0=minimal output, 1=a little, and 2=a lot=current output, 3=debugging output=old `verbose` mode)

2. Guard more of the MPB output with verbosity checks so that, e.g. frequencies aren't outputted in `verbosity<2` mode.

3. Default to `verbosity=1` mode for interactive Python usage like notebooks.

### PR DESCRIPTION
**Pull request by RobinD42**
Fixes #116 

Note that since there is also mpb code (python wrappers and such) in the Meep lib then there will be another PR there that is part of this change.

### CODE REVIEW

**Comment by stevengj** [comment]
The Travis failure on MacOS seems to be new?

**Comment by RobinD42** [comment]
I'm looking into the test failure now.

**Comment by RobinD42** [comment]
The tests pass for me on my machine, so it may be that something has changed on Travis.

**Comment by oskooi** [comment]
The problem was indeed with Travis. After restarting just the failing macOS test, it's passing now.

File: src/matrices/linmin.c [hunk 1]
```diff
@@ -93,12 +94,12 @@ double linmin(double *converged_f, double *converged_df,
 	  }
 
 	  if (*task != 'C') {  /* not converged; warning or error */
-	       if (verbose || *task == 'E')
+	       if ((verbose || mpb_verbosity >= 2 || *task == 'E')  && mpb_verbosity >= 2)
```
Comment (thchr) [inline src/matrices/linmin.c:97]: Shouldn't this have been `mpb_verbosity >= 3`? Same for line 102.
Right now, this gets spammed multiple times for every k-point when calling mpb, which seems undesirable as default behavior.

**Comment by thchr** [comment]
This seems to have added some I-think-unintended spam to the ordinary operation of MPB from the command line (see e.g. the CI logs: https://travis-ci.org/github/NanoComp/mpb/jobs/729108444#L4730)

I've indicated what I think the issue is above; if either @RobinD42 or @stevengj can confirm this is the right fix, I can submit a PR.

**Comment by RobinD42** [comment]
Yes, there is still some adjustments that need to be made when checking the verbosity level. To keep things simpler when migrating to this new scheme I just checked for 2 everywhere. Previously most things were unconditional, or were based on a boolean flag, so the first pass was intended mainly to switch to the new verbosity flags, with the fine-tuning to happen later. 

Any help or suggestions for that fine-tuning is welcome.

**Comment by stevengj** [comment]
Please feel free to submit a PR to make the default less verbose.

**Pull request merged**